### PR TITLE
setup: added react base setup - Issue#9

### LINKED ISSUE #9: UI Repository base setup
## Description
Have a setup working with all dependencies and required directory structure. 

## Acceptance Criteria
- [x] create a base setup for a react app
- [ ] add dependencies for testing
- [x] you should be able to build and run your app

## Definition of Done
- [x] All of the required items are completed.
- [x] Approval by 2 mentor.

## Estimation
1 day

### PR DESCRIPTION
**Pull request by mtreacy002**
### Description

Created boilerplate for React app and merged with the existing bridge-in-tech-web initial commits

Fixes #9 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
run the `npm start` and confirmed the basic react app runs
Screen Shot 2020-06-02 at 7 31 10 pm

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by mtreacy002** [comment]
Update:
@anitab-org/bridgeintech-maintainers, so, I've added the main navbar template (for non-logged in user). 
Screen Shot 2020-06-03 at 2 08 52 pm


Just need to work on the logo and the layout to make it look like the high-fidelity mockups.
Screen Shot 2020-06-03 at 2 05 50 pm

@meenakshi-dhanani, do you want me to complete the main navbar on this issue or should I do that in the next issue for register user ([issue#8](https://github.com/anitab-org/bridge-in-tech-web/issues/8))?

**Comment by meenakshi-dhanani** [comment]
Let's do small issues, let this one just have the base setup, we expand on the nav bar as and when we pick up those stories

**Comment by mtreacy002** [comment]
Ok. so, will what we have so far suffice for the issue to be closed or is there anything else I need to do for this setup?

**Comment by meenakshi-dhanani** [comment]
Should be enough I think. Let me just recheck and approve if all's good.

**Comment by mtreacy002** [comment]
I've just pushed the latest change (removing search field)

File: src/serviceWorker.js
Comment (meenakshi-dhanani) [inline src/serviceWorker.js]: We won't need this just yet. You can remove this

File: public/robots.txt
Comment (meenakshi-dhanani) [inline public/robots.txt]: Don't need this file as well

File: public/manifest.json
Comment (meenakshi-dhanani) [inline public/manifest.json:1]: Don't need this too as of now

Comment (meenakshi-dhanani) [reply public/manifest.json:1]: You can remove the React logos. The AnitaB.org logo should work for us

**Comment by meenakshi-dhanani** [comment]
Refrain from putting all your components in a directory called js. Everything that you are going to write is going to be js. And introduce a directory in the structure, when you find some grouping. eg. listings, details, or programs, company(domain related)

File: public/manifest.json
Comment (mtreacy002) [reply public/manifest.json:1]: should I delete the manifest.json file or just its content related to react logo?

File: public/robots.txt
Comment (mtreacy002) [reply public/robots.txt]: ok. I'll delete this file then 👌

File: src/serviceWorker.js
Comment (mtreacy002) [reply src/serviceWorker.js]: can you please tell me why we don't need service worker? Based on what I read [here](https://stackoverflow.com/questions/49314386/what-is-service-worker-in-react-js) it seems like we're gonna use it a lot, especially for the following features:
```
* ability to intercept and handle network requests
* programmatically managing a cache of responses
* helps cache assets and other files when the user is offline or on slow network
```

I understand we don't need the service worker for this PR specifically, but why do we have to limit ourselves to only this PR? Isn't the purpose of using create-react-app boilerplate is so that we can get the basic stuff it provides that is enough to get us started instead of writing everything from scratch?

**Comment by mtreacy002** [comment]
> Refrain from putting all your components in a directory called js. Everything that you are going to write is going to be js. And introduce a directory in the structure, when you find some grouping. eg. listings, details, or programs, company(domain related)

I agree with you on this. which is why I started creating a folder called `components` under `js` folder

Screen Shot 2020-06-03 at 8 16 52 pm

I'm also planning to group few things later (not sure how, will see as I go) perhaps, a separate folder based by functions, e.g.:
  *  `components` which could have sub folders `navbars`, `forms`, `profiles`, `lists`, `portfolios`
  * `views` or `pages` which could have sub folders like `non-organisation-member`, `organisation-member`, `public`

I probably will have better idea as we move forward. 😉

File: src/routes.js
Comment (meenakshi-dhanani) [inline src/routes.js:1]: router.js will keep all nav related details. You might also need to connect to different components based on the route. I think no need to extract it out to MainNavPublic. Check this out: https://stackoverflow.com/questions/47338077/react-router-v4-navlink-vs-link-benefits

File: public/manifest.json
Comment (mtreacy002) [reply public/manifest.json:1]: added the AnitaB icon instead for tab icon to replace react favicon icon 😂 
Hope this is Ok, @meenakshi-dhanani 
Screen Shot 2020-06-03 at 9 35 05 pm

File: src/routes.js
Comment (mtreacy002) [reply src/routes.js:1]: I'm not sure I'm following you here. Can you please elaborate?

Comment (meenakshi-dhanani) [reply src/routes.js:1]: I mean, you can simple place your NavLinks in the router.js file. The file is going to have routing details, and links only. So you need not place that in a separate component

Comment (mtreacy002) [reply src/routes.js:1]: Ok. will work on it first thing tomorrow morning 😉

Comment (mtreacy002) [reply src/routes.js:1]: @meenakshi-dhanani . just wanna give you an update. I have put back the navbar inside route.js and pushed the changes.
Can you pls now re-review this PR so I can go ahead working on [issue#8](https://github.com/anitab-org/bridge-in-tech-web/issues/8) as this issue#9 is blocking it 😄 ? thanks

File: src/Home.jsx
Comment (mtreacy002) [inline src/Home.jsx:2]: Done, @meenakshi-dhanani . Now Home.jsx sits inside src directory and I removed the js directory. 😉

File: src/index.js
Comment (meenakshi-dhanani) [inline src/index.js]: Remove commented lines. We don't need them

File: src/App.test.js
Comment (meenakshi-dhanani) [inline src/App.test.js]: This is a failing test. We need to fix this

**Comment by meenakshi-dhanani** [comment]
Screen Shot 2020-06-04 at 3 31 45 PM
The logo (AnitabLogo.png) isn't loading, but the icon file (ico) is loading. Does the app look the same on your local too?

File: src/App.test.js
Comment (mtreacy002) [reply src/App.test.js]: in the same PR or open separate issue for this?

Comment (mtreacy002) [reply src/App.test.js]: This is just given by the boilerplate. Should I just remove it?

Comment (mtreacy002) [reply src/App.test.js]: The 3 App files here are just boilerplates. Shall I remove them?

Screen Shot 2020-06-04 at 8 08 32 pm

Comment (meenakshi-dhanani) [reply src/App.test.js]: Yes if you are not using them, remove it.

**Comment by mtreacy002** [comment]
> Screen Shot 2020-06-04 at 3 31 45 PM
> 
> The logo (AnitabLogo.png) isn't loading, but the icon file (ico) is loading. Does the app look the same on your local too?

Yeah, I'm not sure why the logo is not loading. Can you give me a hint? I think I have done the required steps for putting image on there. Or do I need to create a separate .css file specific to navigation bar?
Do you want to create a specific issue for this or just should I just do it here? 
While we're on it, should I fix the MainNavPublic to look like the mockup as I mentioned before [on the post above](https://github.com/anitab-org/bridge-in-tech-web/pull/10#issuecomment-637940360)? 
I left it like that coz I thought you said we should do small issue just for base setup for now. 🤣

File: src/App.test.js
Comment (mtreacy002) [reply src/App.test.js]: Got it. All removed now. Anything else I need to do before I pushed the latest commit?

**Comment by meenakshi-dhanani** [comment]
Let's look at it in the issue we create for the navbar. Sorry so final changes is to remove those unused App files and the test for it and remove commented lines if any. And then I approve for sure

**Comment by mtreacy002** [comment]
> Let's look at it in the issue we create for the navbar. Sorry so final changes is to remove those unused App files and the test for it and remove commented lines if any. And then I approve for sure

Done. I'll push the change now 😉

**Comment by mtreacy002** [comment]
Pushed the changes just now 😉

**Review by meenakshi-dhanani — APPROVED** [review]

**Comment by meenakshi-dhanani** [comment]
@foongminwong can you check if it seems fine to you?

**Review by foongminwong — APPROVED** [review]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done by @meenakshi-dhanani 

2. All possible responses were tested as below:

  * _Homepage_  
    _Screenshot/gif/url_:  
bit-pr-10

    _Expected Result_:  A user should be able to see a simple BIT React base setup + under construction landing page.
    _Actual Result_: Same as above and shown.
    
3. Additional testcases covered: N/A
    
4. Additional Comments: @mtreacy002 
* The logo (AnitabLogo.png) doesn't show up as shown in the GIF, but I was able to see the icon file (ico) on my end (Google Chrome) as shown below:
* bit-pr-10-icon


5. Status of PR Changed to: N/A

6. OS Version: Windows 10

**Comment by foongminwong** [comment]
since @meenakshi-dhanani created [an issue on the logo](https://github.com/anitab-org/bridge-in-tech-web/issues/13) (which can be dealt later), i'll merge this PR

**Pull request merged**
